### ISSUE TITLE
feat: Improve ion-reorder accessibility

### ISSUE THREAD

**Issue by dpalou** [ionic-team/ionic-framework#29296]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already include this feature request, without success.

### Describe the Feature Request

I'm not an accessibility expert so I might be wrong, but I think ion-reorder is not accessible. Some of the things that could be improved are:

- According to [WCAG 2.2 Success Criterion 2.5.7: Dragging Movements (Level AA)](https://www.w3.org/WAI/WCAG22/Understanding/dragging-movements), there should be an alternative using a "single pointer mode of operation without the need for the dexterity required to drag elements". I'm not sure what this means though. maybe there should be some arrows or similar to move the elements with clicks.
- Along with the previous point, maybe users should be able to move the elements using keyboard. I guess that if some arrows are added and they're clickable by keyboard, this point would be fulfilled too.
- When ion-reorder is just an "icon" (not using "Reorder Wrapper"), it would be nice if Ionic automatically calculated the aria-label of ion-reorder based on the item contents (unless it already has an aria-label, aria-labelledby or similar). I think other Ionic components do something similar. Also, I guess the aria-label should also explain how to move the item.
- When using Talkback and selecting the item's label (again, not using "Reorder Wrapper"), Talkback says "double tap to activate" but there is nothing to activate there.
- When an item is repositioned, I guess there should be some message in an "aria-live" element.

### Describe the Use Case

As a visual impaired user, I want to be able to use ion-reorder easily (using Talkback/Voiceover or a keyboard).

### Related Code

This is the template I used, in case it's helpful:

```
<ion-reorder-group [disabled]="dragDisabled" (ionItemReorder)="handleReorder($event.detail)">
    @for (item of items; track $index) {
      <ion-card>
          <ion-item class="ion-text-wrap">
              <ion-label [id]="item.id">
                  {{ item.label }}
              </ion-label>
              <ion-reorder [attr.aria-labelledby]="item.id" slot="end" />
          </ion-item>
      </ion-card>
    }
</ion-reorder-group>
```

**Comment by liamdebeasi** [comment]
Hey there,

Can you provide a full code sample that I can use to reproduce each behavior?

**Comment by dpalou** [comment]
Hi Liam,

if you mean some code for each of the possible improvements, right now I haven't done anything to improve them (besides adding an aria-label to ion-reorder). We might add some elements on top of ion-reorder to fix some of the accessibility issues (I still need to talk about this with our UX), I'll be happy to share the code with you once I'm done.

If you mean some code to reproduce the problems, our app is quite complex so it wouldn't be easy for you to replicate it with our code. But basically adding ion-reorder anywhere would have those accessibility problems.

Thank you!

**Comment by FabianGutierrezOpengov** [comment]
@liamdebeasi our accessibility team is also looking for this, at least the ability of using keyboard to reorder.  I provided a quick sample where I added tabindex to the ion-reorder element, but this doesnt feel like a something to "fix" but an important missing accessibility feature as kindly established by @dpalou 

https://stackblitz.com/edit/angular-cc5twc-xutyk1?file=src%2Fapp%2Fexample.component.html

In any case if you need more feedback, happy to help.

have a good day :)

**Comment by liamdebeasi** [comment]
Hey there,

I no longer work on Ionic Framework, so I won't be much help in moving this along. Thanks for creating this demo though! This will be helpful for the dev team to better understand this feature request.
